spec/controllers/verify/usps_controller_spec.rb

```diff
@@ -58,37 +59,53 @@
         expect(response).to redirect_to verify_review_path
         expect(subject.idv_session.address_verification_mechanism).to eq :usps
       end
+
+      it 'creates an initial :usps_mail_sent event' do
+        put :create
+
+        expect(user_event_types).to eql(%w[usps_mail_sent])
+      end
     end
 
     context 'resending a letter' do
       let(:has_pending_profile) { true }
       let(:pending_profile) { create(:profile, phone_confirmed: false) }
+      let(:usps_confirmation_maker) { instance_double(UspsConfirmationMaker) }
 
       before do
         stub_sign_in(user)
         stub_decorated_user_with_pending_profile(user)
         allow(user.decorate).to receive(:needs_profile_usps_verification?).and_return(true)
-      end
 
-      it 'calls the UspsConfirmationMaker to send another letter and redirects' do
         pii = { first_name: 'Samuel', last_name: 'Sampson' }
         pii_cacher = instance_double(Pii::Cacher)
         allow(pii_cacher).to receive(:fetch).and_return(pii)
         allow(Pii::Cacher).to receive(:new).and_return(pii_cacher)
 
         session[:sp] = { issuer: '123abc' }
 
-        usps_confirmation_maker = instance_double(UspsConfirmationMaker)
         allow(UspsConfirmationMaker).to receive(:new).
           with(pii: pii, issuer: '123abc', profile: pending_profile).
           and_return(usps_confirmation_maker)
+      end
 
+      def put_create
         expect(usps_confirmation_maker).to receive(:perform)
-
         put :create
+      end
+
+      it 'calls the UspsConfirmationMaker to send another letter and redirects' do
+        put_create
 
         expect(response).to redirect_to verify_come_back_later_path
       end
+
+      it 'creates a subsequent :usps_mail_sent event' do
```

Comment:
Sorry, I was wrong about a bug being present here. I was misreading the code. In order to make sure that an additional event gets created for a user who requests a subsequent letter, I think the best place to test that is on line 149 of this test:
https://github.com/18F/identity-idp/blob/master/spec/features/saml/loa3_sso_spec.rb#L149

by adding the following expectation:
```ruby
expect(user.events.usps_mail_sent.size).to eq 2
```

I would do that instead of this controller spec. Thoughts?

Reply:
I agree :+1: